packages/authentication/src/metadata.ts

```diff
@@ -0,0 +1,15 @@
+// Copyright IBM Corp. 2013,2017. All Rights Reserved.
+// Node module: loopback
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import * as assert from 'assert';
+import 'reflect-metadata';
+
+export function authenticate(controllerClass: Object, methodName: string) {
+  Reflect.defineMetadata('loopback:authenticate', true, controllerClass, methodName);
```

Comment:
I believe this implementation supports additional decorator usages:

```ts
// class-level decoration
@authenticate()
class TestClass {
  // parameter-level decoration
  foo(@authenticate() id: string): string {
    return id;
  }
}
```

Assuming we are not going to support these two use cases, then `@authenticate()` should throw a helpful error when used incorrectly. This should be covered by the tests too.